src/tokens.js

```diff
@@ -112,45 +112,72 @@ const tokens = {
    * @param {string} config Token configuration string
    * @param {string} uid Uid to check if matches with uid from config (optional)
    *
-   * @return {Object} {success: boolean, message: in case of failure, tokenData: object with token data in case of success}
+   * @return {Promise} Promise that resolves when validation finishes. Resolves with tokenData {uid, name, symbol} and reject with error message
    *
    * @memberof Tokens
    * @inner
    */
   validateTokenToAddByConfigurationString(config, uid) {
-    const tokenData = this.getTokenFromConfigurationString(config);
-    if (tokenData === null) {
-      return {success: false, message: 'Invalid configuration string'};
-    }
-    if (uid && uid !== tokenData.uid) {
-      return {success: false, message: `Configuration string uid does not match: ${uid} != ${tokenData.uid}`};
-    }
+    const promise = new Promise((resolve, reject) => {
+      const tokenData = this.getTokenFromConfigurationString(config);
+      if (tokenData === null) {
+        reject('Invalid configuration string');
```

Comment:
We have issue #22 open so that we reject the error object, not just the message. This way, clients using the lib can catch specific errors. I think we can start doing in the parts of the lib we are refactoring.

If you expose the error, you can later test it instead of expecting a generic string `rejects.toEqual(expect.any(String))`. We do something like this in the test 'Insufficient funds' for `__tests__/token_utils.js`.

We would need to create additional error classes, like TokenValidationError and so on.

Reply:
Agreed! Now the error is being rejected

Reply:
I did the test a little bit different than the 'Insufficient funds' one, do you think is it a problem how it's done now? 

https://github.com/HathorNetwork/hathor-wallet-lib/pull/26/files#diff-616b0e865ab989f74898b8625441a9d3R183

Reply:
Yeah, looks even better than the other code.